### PR TITLE
added visual regression code example using Smart ui from LambdaTest

### PR DESCRIPTION
**Pull request by mfaisalkhatri**

### CODE REVIEW

File: testing-modules/selenium/src/test/java/com/baeldung/selenium/visualregression/pages/CameraProductPage.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import java.text.MessageFormat;
+
+public class CameraProductPage {
+    private final WebDriver driver;
+
+    public CameraProductPage(final WebDriver driver) {
+        this.driver = driver;
+    }
+
+    private static final String SCREEN_NAME = "Camera-Product-Page";
```
Comment (acerasoni) [inline testing-modules/selenium/src/test/java/com/baeldung/selenium/visualregression/pages/CameraProductPage.java:15]: Let's put this at the top of the class

File: testing-modules/selenium/src/main/java/com/baeldung/selenium/visualregression/DriverManager.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+import java.util.HashMap;
+
+import static java.text.MessageFormat.format;
+
+public class DriverManager {
+
+    private WebDriver driver;
+
+    public void startChromeInCloud()  {
+        final String LT_USERNAME = System.getenv("LT_USERNAME");
```
Comment (acerasoni) [inline testing-modules/selenium/src/main/java/com/baeldung/selenium/visualregression/DriverManager.java:20]: We typically don't make variables final in our code examples (even though I agree they should be in production-grade code)

Comment (acerasoni) [inline testing-modules/selenium/src/main/java/com/baeldung/selenium/visualregression/DriverManager.java:20]: Also, despite the environment variable being in all caps, the variables names should be camel-cased

File: testing-modules/selenium/testng.xml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    <test name="Visual Regression Test using Smart UI from LambdaTest">
+        <classes>
+            <class name="com.baeldung.selenium.visualregression.tests.SeleniumVisualRegressionLiveTest">
+                <methods>
+                    <include name="whenActualImageIsDifferentFromBaseline_thenItShouldShowDifference"/>
+                    <exclude name="whenActualImageIsSameAsBaseline_thenItShouldNotShowAnyDifference"/>
+                </methods>
+            </class>
+        </classes>
+    </test> <!-- Test -->
```
Comment (acerasoni) [inline testing-modules/selenium/testng.xml:13]: Can we remove  and ?

File: testing-modules/selenium/src/main/java/com/baeldung/selenium/visualregression/DriverManager.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    }
+
+    public WebDriver getDriver() {
+        return this.driver;
+    }
+
+    public void quitDriver() {
+        this.driver.quit();
+    }
+
```
Comment (acerasoni) [inline testing-modules/selenium/src/main/java/com/baeldung/selenium/visualregression/DriverManager.java:67]: Could we clean up blank lines at the end of the classes?

File: testing-modules/selenium/pom.xml [hunk 5]
```diff
@@ -56,8 +56,8 @@
     </build>
 
     <properties>
-        <testng.version>6.10</testng.version>
-        <selenium-java.version>4.8.3</selenium-java.version>
+        <testng.version>7.9.0</testng.version>
```
Comment (acerasoni) [inline testing-modules/selenium/pom.xml:59]: Have we checked that increasing these dependencies doesn't break the other code examples?

File: testing-modules/selenium/testng.xml (hunk 3 above)
Comment (mfaisalkhatri) [reply testing-modules/selenium/testng.xml:13]: Should I remove this testng.xml file?

File: testing-modules/selenium/pom.xml (hunk 5 above)
Comment (mfaisalkhatri) [reply testing-modules/selenium/pom.xml:59]: Reran all the tests, most of the tests ran successfully. The following observations were found : 
1. The following tests inside the package >> “stale” were throwing error  - “org.openqa.selenium.SessionNotCreatedException: Could not start a new session. Response code 500. Message: session not created: This version of ChromeDriver only supports Chrome version 114.” Upgraded the WebDriverManager dependency version to latest version(5.7.0) and it worked fine:
StaleElementReferenceLiveTest
RobustWebElementLiveTest

2. The following test in SeleniumPageObjectLiveTest class fails as it couldn’t locate the WebElement with locator - “.page-title”
givenHomePage_whenNavigate_thenShouldBeInStartHere()

3. The following test in SeleniumPageObjectLiveTest class fails as it couldn’t locate the WebElement with locator - “.page-header”
givenAboutPage_whenNavigate_thenTitleMatch()

The following test in TakeScreenShotSeleniumLiveTest class fails as it couldn’t locate the WebElement with locator - “#hplogo”
whenGoogleIsLoaded_thenCaptureLogo()

File: testing-modules/selenium/src/main/java/com/baeldung/selenium/visualregression/DriverManager.java (hunk 2 above)
Comment (mfaisalkhatri) [reply testing-modules/selenium/src/main/java/com/baeldung/selenium/visualregression/DriverManager.java]: updated

Comment (mfaisalkhatri) [reply testing-modules/selenium/src/main/java/com/baeldung/selenium/visualregression/DriverManager.java]: updated

File: testing-modules/selenium/src/main/java/com/baeldung/selenium/visualregression/DriverManager.java (hunk 4 above)
Comment (mfaisalkhatri) [reply testing-modules/selenium/src/main/java/com/baeldung/selenium/visualregression/DriverManager.java]: removed the blank lines.

File: testing-modules/selenium/src/test/java/com/baeldung/selenium/visualregression/pages/CameraProductPage.java (hunk 1 above)
Comment (mfaisalkhatri) [reply testing-modules/selenium/src/test/java/com/baeldung/selenium/visualregression/pages/CameraProductPage.java]: done

File: testing-modules/selenium/testng.xml (hunk 3 above)
Comment (acerasoni) [reply testing-modules/selenium/testng.xml:13]: Apologies, my comment didn't format properly. I meant to say, can we remove the  and  inline comments?

**Pull request merged**